src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java

```diff
@@ -1,26 +1,102 @@
+        return errors;
+    }
+
+    private boolean validateServiceAddressSameAsPrincipalAddress(Boolean same, Errors errors, String loggingContext) {
+        String location = getEntityFieldName(EntityDto.IS_SERVICE_ADDRESS_SAME_AS_PRINCIPAL_ADDRESS_FIELD);
+        return UtilsValidators.validateNotNull(same, location, errors, loggingContext);
+    }
+
+    private boolean validateEmail(String email, Errors errors, String loggingContext) {
+        String location = getEntityFieldName(EntityDto.EMAIL_PROPERTY_FIELD);
+        return StringValidators.validateStringNotBlank(email, location, errors, loggingContext)
+                && StringValidators.validateMaxLength(email, 250, location, errors, loggingContext)
+                && StringValidators.validateEmail(email, location, errors, loggingContext);
+    }
+
+    private boolean validateLegalForm(String legalForm, Errors errors, String loggingContext) {
+        String location = getEntityFieldName(EntityDto.LEGAL_FORM_FIELD);
+        return StringValidators.validateStringNotBlank(legalForm, location, errors, loggingContext)
+                && StringValidators.validateMaxLength(legalForm, 4000, location, errors, loggingContext)
+                && StringValidators.validateCharacters(legalForm, location, errors, loggingContext);
+    }
+
+    private boolean validateLawGoverned(String lawGoverned, Errors errors, String loggingContext) {
+        String location = getEntityFieldName(EntityDto.LAW_GOVERNED_FIELD);
+        return StringValidators.validateStringNotBlank(lawGoverned, location, errors, loggingContext)
+                && StringValidators.validateMaxLength(lawGoverned, 4000, location, errors, loggingContext)
+                && StringValidators.validateCharacters(lawGoverned, location, errors, loggingContext);
+    }
+
+    private boolean validatePublicRegisterName(String publicRegisterName, Errors errors, String loggingContext) {
+        String location = getEntityFieldName(EntityDto.PUBLIC_REGISTER_NAME_FIELD);
+        return StringValidators.validateStringNotBlank(publicRegisterName, location, errors, loggingContext)
+                && StringValidators.validateMaxLength(publicRegisterName, 4000, location, errors, loggingContext)
+                && StringValidators.validateCharacters(publicRegisterName, location, errors, loggingContext);
+    }
+
+    private boolean validateRegistrationNumber(String registrationNumber, Errors errors, String loggingContext) {
+        String location = getEntityFieldName(EntityDto.REGISTRATION_NUMBER_FIELD);
+        return StringValidators.validateStringNotBlank(registrationNumber, location, errors, loggingContext)
+                && StringValidators.validateMaxLength(registrationNumber, 32, location, errors, loggingContext)
+                && StringValidators.validateCharacters(registrationNumber, location, errors, loggingContext);
+    }
+
+
     private String getEntityFieldName(String fieldName) {
```

Comment:
Appreciate you didn't change this line/method but would be nice to bring method name in line with the one that's more sensibly named in the `AddressDtoValidator` class, so `getQualifiedEntityFieldName`.

Reply:
Ok updated this as well.